### DISCUSSION TITLE
Remove the blank bar at the bottom of helix when it is not being used, and improve the discoverability of modal editing

### DISCUSSION THREAD

**Discussion by godalming123** [helix-editor/helix#10818] [Ideas]
There is a bar at the bottom of helix that is usually blank. When it is not blank it either stores the last status message (EG `Can't save with no path set!`), prompts (EG `Search:`), or the series of keys that the user has pressed (EG `9<space>g`). I think that the series of keys that the user has pressed can be moved into the key prompt menu:
```
┌ <9> * <space> <w>indow ───────────┐
│ w, C-w   Goto next window         │
│ s, C-s   Horizontal bottom split  │
│ v, C-v   Vertical right split     │
│ t, C-t   Transpose splits         │
│ ...                               │
└───────────────────────────────────┘
```
The key prompt menu would probably also want to be moved to the top of the screen so that it's title is always in the same position no matter how many items there are in it. Although, alternatively you could put the `<9> * <space> <w>indow` on the bottom of the key prompt menu. Then, when their is no status message, the blank bar could be hidden, and when a status message (or prompt) is shown, the status line would move up one line to make space for the status message or prompt.

This would make it easier for a new user to discover how modal editing works by more directly linking the keys pressed with what new key presses will do, and explicitly showing how numbers repeat an instruction. It also removes the blank bar at the bottom when it is not in use which saves space. As well as this, it would allow users to see which key bindings there are in normal mode by pressing 1, and then the menu would come up:
```
┌ <1> * ────────────────────┐
│ [Normal mode keybindings] │
└───────────────────────────┘
```

In order to make space for the key-prompt menu if it goes on the top, the LSP diagnostics could move to the bottom right. For users that have the key-prompt menu disabled in their config, I think that  the key presses could become a status-bar component which means that a user could move them to a different side, or the status bar could move up, and the key presses could move to the space below the status bar.

**Comment by nik-rev** [comment]
You may be interested in https://github.com/helix-editor/helix/pull/12204
